packages/db/src/translate/granules.ts

```diff
@@ -95,7 +95,7 @@ export const translatePostgresGranuleToApiGranule = async ({
     endingDateTime: granulePgRecord.ending_date_time?.toISOString(),
     error: granulePgRecord.error,
     execution: executionUrls[0] ? executionUrls[0].url : undefined,
-    files: files.length > 0 ? files.map((file) => translatePostgresFileToApiFile(file)) : undefined,
+    files: files.map((file) => translatePostgresFileToApiFile(file)),
```

Comment:
Need to understand why we were returning `undefined` here. It changed in this PR: https://github.com/nasa/cumulus/pull/2426/files#diff-cb9b3877757df3f7f8ff3e249d9aae53c0d96be9e32c7cab69bfd8af1ba13b9aR97

Reply:
Returning an empty array like we were before the above change makes life easier so I'd prefer that. It's probable that this was returning `undefined` and, because we weren't removing undefined/null properties this is a minor change...

Reply:
We also _have to_ return an empty array here I think. These changes are based on our assertion that `files: undefined` and `files: []` are NOT the same thing.

Reply:
That PR was related to updating the reconciliation reports to read from postgres.    Does this set of changes make the files comparison in those reports not work somehow?

Reply:
I was thinking 'no' originally because it seemed like it only impacted unit test granule comparisons but you may be right. In `reconciliationReport.js` we're comparing `apiGranule.files` to `undefined` explicitly: https://github.com/nasa/cumulus/pull/2426/files#diff-35c1d9ed4359fe09e9216bec5e5de87269cf68d7c8ce7ff000b484f6a2c7f610R254-R255

The unit tests for rec reports pass with my change, which is unfortunate. It looks like we're not actually creating PG/ES Granule records in tests but just arbitrary objects so it's not testing the translate logic.

Also concerning is an explicit comparison of ES granule files to `undefined` in the link above:

`const esGranulesHasFiles = esGranule.files !== undefined;`

I was under the impression based on my tests that, if an esGranule doesn't have files it'll return an empty array, not undefined. More to look at.

Reply:
😦  thanks for looking into that further.

Reply:
So I reverted the change here in the translate method. We're now returning `files: undefined` again if there are no files. The argument _against_ returning `files: undefined` is that the response then differs between Elastic/Dynamo and PG. See this comment: https://github.com/nasa/cumulus/pull/3082#discussion_r967341644

The upside is that we don't have to modify granule factories or reconciliation report logic if we continue returning `undefined`. That, to me, was convincing enough to not make this change right now. 

I added a couple more tests here to make it easier to validate the files behavior. e.g. https://github.com/nasa/cumulus/pull/3082/files#diff-e4b95ee0e3b92433a8e670895c1e8da32d083c3ca64e31761d4bb353cbb791a7R617